src/recon.rs

```diff
@@ -4469,22 +4470,42 @@ pub(crate) unsafe fn rav1d_filter_sbrow_deblock_cols<BD: BitDepth>(
     }
     let y = sby * f.sb_step * 4;
     let ss_ver = (f.cur.p.layout as c_uint == Rav1dPixelLayout::I420 as c_int as c_uint) as c_int;
-    let p: [*mut BD::Pixel; 3] = [
-        (f.lf.p[0] as *mut BD::Pixel)
-            .offset((y as isize * BD::pxstride(f.cur.stride[0] as usize) as isize) as isize),
-        (f.lf.p[1] as *mut BD::Pixel).offset(
-            (y as isize * BD::pxstride(f.cur.stride[1] as usize) as isize >> ss_ver) as isize,
+    let ss_hor = (f.cur.p.layout as c_uint != Rav1dPixelLayout::I444 as c_int as c_uint) as c_int;
+
+    let datay_stride = BD::pxstride((*f).cur.stride[0] as usize) as isize;
+    let datay_width = (*f).cur.p.w;
+    let datay_height = (*f).cur.p.h;
+    let datay_diff = (datay_height - 1) as isize * datay_stride;
+    let datauv_stride = BD::pxstride((*f).cur.stride[1] as usize) as isize;
+    let datauv_width = datay_width >> ss_hor;
+    let datauv_height = datay_height >> ss_ver;
+    let datauv_diff = (datauv_height - 1) as isize * datauv_stride;
```

Comment:
Could you add `_`s after the `data` for these variables?

Reply:
You could also just do all these calculations for `p` and `p_offset` inside an inner block and then drop the `data*` prefixes of all these variables since the variables have a tight scope.

Reply:
Moved it to an inner block